pyqtgraph/graphicsItems/ViewBox/ViewBox.py

```diff
@@ -353,8 +348,20 @@ def setMouseMode(self, mode):
         if mode not in [ViewBox.PanMode, ViewBox.RectMode]:
             raise Exception("Mode must be ViewBox.PanMode or ViewBox.RectMode")
         self.state['mouseMode'] = mode
+        self._add_rectangle_selection()
+
         self.sigStateChanged.emit(self)
 
+    def _add_rectangle_selection(self):
+        if self.reScaleBox is None and self.state['mouseMode'] == ViewBox.RectMode:
```

Comment:
`self.reScaleBox` should be a typo for `self.rbScaleBox`?
As it is, a new `QGraphicsRectItem` is instantiated each time `mouseMode` changes to `RectMode`

Reply:
Yes, that was a typo 🤦‍♂️